src/content/docs/ra-primer/ra-in-action.mdx

```diff
@@ -390,5 +390,5 @@ From the resulting relation, we can for example learn that:
 
 > Peter Sellers and Jennifer Jason Leigh are related in that Sellers worked with Stanley Kubrik, who also worked with Shelley Duval, who in turn worked with Robert Altman, who also worked with Sellers!
 
-Extending our query required only two invokations of the extracted methods, and a modification of the final `restrict`. This kind of modularity is in general impossible to achieve in SQL, unless you succumb to concatenating query strings. This might seem like an unfair comparison, since we are using a general-purpose programming language. However, even [Object-Relational Mappings](https://en.wikipedia.org/wiki/Object%E2%80%93relational_mapping) and other database wrappers have problems expressing this kind of composition in a natural way. As we have noted earlier, this is because they use SQL as the underlying query model, and SQL is inherently poor at composition.
+Extending our query required only two invokations of the extracted methods, and a modification of the final `restrict`. This kind of modularity is in general impossible to achieve in SQL, unless you succumb to concatenating query strings. This might seem like an unfair comparison, since we are using a general-purpose programming language. However, even [Object-Relational Mappings](https://en.wikipedia.org/wiki/Object%E2%80%93relational_mapping) and other database wrappers have problems expressing this kind of composition in a natural way. As we have noted earlier, this is because they use SQL abstract syntax trees as the underlying query model, and SQL's syntax is inherently poor at composition.
```

Comment:
Would like to discuss this a bit more. Your suggestion introduces new concepts without explaining them. If we just say "SQL" without specifying, it means SQL as a whole, and to me it seems SQL is first and foremost an (abstract) syntax. For people who know a bit more, they might fill in the blanks of "underlying query model" and understand which aspects of SQL is limiting.

So - in the interest of keeping things simple, I'd prefer not to change this. But if you think it's misleading as it stands, please explain more!

Reply:
SQL is at least as composable as relational algebra, since "SELECT ... FROM (SELECT ... FROM ...))" is supported. A naive translation of relational algebra to SQL is super straightforward.

The thing is that no one wants such a naive translation, because of performance and readability reasons. But it has nothing to do with SQL, and all to do with the SQL implementations, optimizers in particular... and with the fact that usual libraries (Arel, Sequel, etc.) represent SQL ASTs, not relations.

Possibly a detail for the reader, definitely not for me. Especially because I don't like saying something wrong about SQL. There are enough bad things in SQL to invent new ones.